core/client/fs/src/main/java/alluxio/client/file/cache/store/RocksPageStoreDir.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.client.file.cache.store;
+
+import static com.google.common.base.Preconditions.checkState;
+
+import alluxio.client.file.cache.PageId;
+import alluxio.client.file.cache.PageInfo;
+import alluxio.client.file.cache.PageStore;
+import alluxio.client.file.cache.evictor.CacheEvictor;
+
+import com.google.common.collect.Streams;
+import org.rocksdb.RocksIterator;
+
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+import java.util.function.Consumer;
+import javax.annotation.Nullable;
+
+/**
+ * Represent the dir and file level metadata of a rocksdb page store.
+ */
+public class RocksPageStoreDir extends QuotaManagedPageStoreDir {
+
+  private final PageStoreOptions mPageStoreOptions;
+
+  private RocksPageStore mPageStore;
```

Comment:
I'm a little confused with the class hierarchy. It seems that a `PageStoreDir` owns and manages a `PageStore`, and external code only interacts with `PageStoreDir` directly. Is this correct? If so, it's confusing to have such names, as a `PageStoreDir` feels like a part of a `PageStore`, not the other way around.

Reply:
PageStoreDir is managing the file(Block) level metadata and capacity of each dir.  As you said, the external code (block worker) only interacts with the page store dir to collect an aggregated view of each dir.  

I put PageStoreDir as a parent of PageStore because (1) the PageStoreDir controls the lifecycle of the page store, the page store could be reset/restore by the dir.  (2) On the worker level, we could hide the impl of page, the only model we expose is the dir which is aline with the tiered store.